tests/Kernel/CallForProposalsRenderTest.php

```diff
@@ -0,0 +1,280 @@
+    'oe_content_departments_field',
+    'oe_content_call_proposals',
+    'oe_theme_content_call_proposals',
+    'datetime_testing',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp(): void {
+    parent::setUp();
+
+    $this->installConfig([
+      'composite_reference',
+      'oe_content_reference_code_field',
+      'oe_content_departments_field',
+      'oe_content_call_proposals',
+      'oe_theme_content_call_proposals',
+    ]);
+
+    module_load_include('install', 'oe_content');
+    oe_content_install();
+
+    // Set current user to UID 1, so that by default we can access everything.
+    $account = User::load(1);
+    $this->setCurrentUser($account);
+  }
+
+  /**
+   * Test a call for proposals being rendered as a teaser.
+   */
+  public function testTeaser(): void {
+    // Freeze the time at a specific point.
+    $static_time = new DrupalDateTime('2020-02-17 14:00:00', DateTimeItemInterface::STORAGE_TIMEZONE);
+
+    $publication_date = (clone $static_time)->modify('- 5 days');
+    $opening_date = (clone $static_time)->modify('- 3 days');
+    $deadline_date = (clone $static_time)->modify('+ 3 days');
+
+    /** @var \Drupal\Component\Datetime\TimeInterface $datetime */
+    $time = $this->container->get('datetime.time');
+    $time->freezeTime();
+    $time->setTime($static_time->getTimestamp());
+
+    // Create a Call for proposals node.
+    /** @var \Drupal\node\Entity\Node $node */
+    $values = [
+      'type' => 'oe_call_proposals',
+      'title' => 'Test Call for proposals node',
+      'oe_publication_date' => [
+        'value' => $publication_date->format('Y-m-d'),
+      ],
+      'oe_call_proposals_opening_date' => [
+        'value' => $opening_date->format('Y-m-d'),
+      ],
+      'oe_call_proposals_deadline' => [
+        'value' => $deadline_date->format(DateTimeItemInterface::DATETIME_STORAGE_FORMAT),
+      ],
+      'oe_call_proposals_model' => 'single_stage',
+      'oe_call_proposals_funding' => ['http://publications.europa.eu/resource/authority/corporate-body/ACM'],
+      'oe_reference_code' => 'Call for proposals reference',
+      'uid' => 0,
+      'status' => 1,
+    ];
+    $node = Node::create($values);
+    $node->save();
+
+    $build = $this->nodeViewBuilder->view($node, 'teaser');
+    $html = $this->renderRoot($build);
+
+    // Check Open status label and background.
+    $assert = new ListItemAssert();
+    $deadline_date->setTimeZone(new \DateTimeZone('Australia/Sydney'));
+    $expected_values = [
+      'title' => 'Test Call for proposals node',
+      'meta' => 'Call status: Open',
+      'image' => NULL,
+      'additional_information' => [
+        new PatternAssertState(new FieldListAssert(), [
+          'items' => [
+            [
+              'label' => 'Reference',
+              'body' => 'Call for proposals reference',
+            ], [
+              'label' => 'Opening date',
+              'body' => $opening_date->format('j F Y'),
```

Comment:
When setting expected values we rather set them as the expected strings directly instead of generating them from properties.